### DISCUSSION TITLE
lag time in model

### DISCUSSION THREAD

**Discussion by filippo-ciceri** [nlmixrdevelopment/nlmixr#595] [Q&A] [answered]
Hi,

I am just wondering how to incorporate lagtime into the model.
The `lag`, `alag` commands seem to work only when referenced to the central compartment (`lag(central) <- lagtime`), instead the depot one (as shown [here](https://nlmixrdevelopment.github.io/RxODE/articles/RxODE-syntax.html)).

Also, which are the unit of measure of the estimated parameter? Hours?

Code example here below:

```
two.cmt.allo.lag <- function() {
  ini({

    tk01 <- log(1)
    tcl <- log(500)
    tv <- log(5e+004)
    tk12 <- log(1)
    tk21 <- log(1)
    
    lagtime <- log(50)
    
    eta.cl ~ 0.5
    eta.v ~ 0.5
    prop_err <- c(0,0.5)
    
  })
  model({
    
    D <- Dose
    s <- Study
    
    cl <- exp(tcl + eta.cl) * ((WT/70) ** 0.75)
    v <- exp(tv + eta.v) * (WT/70)
    
    k01 <- exp(tk01) 
    k10 <- cl/v
    k12 <- exp(tk12)
    k21 <- exp(tk21) 
    
    d/dt(cent) = k10 * depot - (cl/v + k12) * cent + k21 * per
    d/dt(depot) = -k01 * depot
    d/dt(per) = k12 * cent - k21 * per
    
    lag(cent) = exp(lagtime)
    
    cp = cent/v
    cp ~ prop(prop_err)
    
  })
}

PKmodel_12 <- nlmixr(two.cmt.allo.lag, data.modelling, est="focei")
```

Thank you very much for the help!! 
`nlmixr` has been so far invaluable for my work! 

Filippo

**Comment by mattfidler** [comment, marked as answer]
Hi @filippo-ciceri `lag(cent)` or `alag(cent)` should be correct.  The time units of the lag time would be the same time units as the input time.

**Comment by filippo-ciceri** [comment]
Just to be 100% clear: I am talking about lag time for absorption of the compound. For this reason I am a bit confused that `lag(central` is correct instead of `lag(depot)`.

**Comment by mattfidler** [comment]
I thought you were talking about syntax.  Most often oral absorption has a lag.  So a lag time on depot would make more sense.

**Comment by filippo-ciceri** [comment]
Thank you. I was a bit confused because the model reaches a good fit with `lag(cent)`, however I was expecting `lag(depot)` to be more appropriate.

**Reply by mattfidler** [reply]
Well, `lag()` will only affect doses in whatever compartment you use.  The `saem` algorithm may allow this to be optimized though it doesn't make much sense in the model itself.

**Reply by filippo-ciceri** [reply]
Yes, I forgot to mention that I was modeling an oral dosing. This is why I thought `lag(depot)` was the better option.
The second part of your answer is a bit more obscure. What exactly can be optimized using the `saem` algorithm?
